mycore-xeditor/src/main/java/org/mycore/frontend/xeditor/MCRXEditorTransformer.java

```diff
@@ -107,9 +114,25 @@ public void addNamespace(String prefix, String uri) {
     }
 
     public void readSourceXML(String uri) throws JDOMException, IOException, SAXException, TransformerException {
+        String uriRe = editorSession.replaceParameters(uri);
+
+        if (!validateURI(uriRe)) {
+            throw new TransformerException("URI '" + uriRe + "' contains illegal uri scheme");
+        }
+
         editorSession.setEditedXML(uri);
     }
 
+    protected boolean validateURI(String uri) {
+        for (String scheme : MCRXEditorTransformer.DISALLOWED_URI_SCHEMAS) {
+            if (uri.contains(scheme)) {
```

Comment:
startsWith()?

Reply:
Does not catch xslStyle:xeditor-preprocessor:callJava:org.mycore.common.xml.MCRXMLFunctions:foo:bar